aws-rds-dbinstance/src/test/java/software/amazon/rds/dbinstance/TranslatorTest.java

```diff
@@ -93,6 +93,20 @@ public void test_modifyDbInstanceRequest_isRollback_IncreaseAllocatedStorage() {
         assertThat(request.allocatedStorage()).isNull();
     }
 
+    @Test
+    public void test_modifyDbInstanceRequest_isRollback_NoAllocatedStorage() {
+        final ResourceModel previousModel = RESOURCE_MODEL_BLDR()
+                .allocatedStorage(null)
+                .build();
+        final ResourceModel desiredModel = RESOURCE_MODEL_BLDR()
+                .allocatedStorage(null)
+                .storageType(STORAGE_TYPE_IO1)
```

Comment:
is this required in the test?

Reply:
Yes, max() function only involved when it is a ProvisionedIoStorage.
https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/blob/master/aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java#L496